saleor/graphql/giftcard/mutations.py

```diff
@@ -87,38 +89,46 @@ class Meta:
 
     @classmethod
     def clean_input(cls, info, instance, data):
+        cleaned_input = super().clean_input(info, instance, data)
+
+        # perform only when gift card is created
         if instance.pk is None:
-            data["code"] = generate_promo_code()
+            cleaned_input["code"] = generate_promo_code()
+            cls.set_created_by_user(cleaned_input, info)
 
-        cleaned_input = super().clean_input(info, instance, data)
-        cls.clean_expiry_settings(cleaned_input)
-        cls.clean_balance(cleaned_input, instance.currency)
+        cls.clean_expiry_settings(cleaned_input, instance)
+        cls.clean_balance(cleaned_input, instance)
 
-        # TODO: send an email to the customer from user_email field in SALEOR-3901
-        data.get("user_email", None)
+        cleaned_input.get("user_email", None)
```

Comment:
Is this still TODO? We're not doing anything with the email.

Reply:
Yes, it will be done later. It's not in scope at the moment.

Reply:
👍